### ISSUE TITLE
Create multiple localization files

### ISSUE THREAD

**Issue by EdStepashkin** [localizely/flutter-intl-intellij#6]
I would like to create multiple localization files . 
For example: feature_one_en.arb, feature_two_en.arb  etc.
It is very useful when I want to split localization by features

**Comment by aleksakrstic** [comment]
Hi @EdStepashkin 
Thanks for suggesting this enhancement.

I agree this would be a nice feature, but unfortunately with low priority for implementation.

How do you plan to ensure you have unique keys among those multiple ARB files?

Instead, would using prefixes for keys be good enough solution, like `pageHome...`, `pageMyProfile...`, as an alternative to multiple files?

**Comment by EdStepashkin** [comment]
I think there are several approaches :
1.  Throw duplicate resource error. Transfer responsibility to developers
2. Using prefixes
3. Using  inner class  for each files (feature)
Also you can add in configuration file strategy of work with multiple files. Everyone will be able to choose the most suitable for themselves.

Thank you for considering the enhancement.

**Comment by aleksakrstic** [comment]
@EdStepashkin Thanks for the proposal.
However, don't expect this soon to be implemented.

**Comment by vsxed** [comment]
@aleksakrstic the possibility to (automatically) add a prefix would be a pretty nice start.

**Comment by evansherry** [comment]
Is there a source code for the plugin? I want to try to modify to meet my needs.
It’s very convenient for me to use the prefix, just modify the `Intl.message` name parameter `(prefix+name)`
```
_notInlinedMessages(_) => <String, Function> {
     "prefix+name": MessageLookupByLibrary.simpleMessage("...")
}
```
It would be nice to provide the optional configuration of prefixName in yaml

**Comment by aleksakrstic** [comment]
Hi @evansherry @EdStepashkin @vsxed 
I would say this enhancement is blocked by https://github.com/dart-lang/intl & https://github.com/dart-lang/intl_translation libraries for the same reason as https://github.com/localizely/flutter-intl-intellij/issues/22 is blocked

@evansherry That part of the code is generated by https://github.com/dart-lang/intl_translation

We will observe where Dart goes with internationalization with Intl library.

**Comment by foreg** [comment]
Hi!
Any updates on this? As far as I know, intl_translation has the following feature since 0.17.5

> Allow multiple ARB files with the same locale and combine their translations.

**Comment by aleksakrstic** [comment]
This would include many changes.
Postponing consideration for few months.
Anyway, feel free to share your motivation why would you use multiple files and how would you split the string keys among them.

**Comment by casvanluijtelaar** [comment]
this feature is absolutely crucial for us. our applications are always split up into several modules (packages) having to declare everything in an applications core and passing it into every single module is a huge time sink and contradicts our modular setup.

**Comment by axel-afonso-blanco** [comment]
Hi!
Any updates on this?
We face the same problems as @casvanluijtelaar, and would hugely benefit from this enhancement.

Thank you very much for considering this enhancement.

**Comment by alxmoroz** [comment]
The reason for the request becomes easier to understand if you imagine what would happen if all the code were also in one file.

**Comment by cedvdb** [comment]
We are currently using multiple `.arb` files for each context of the application. About 20 arb files.

 - For multiple modules it's totally possible currently. However it requires to have a package for each feature or subset of features which can be annoying for quickly navigating between features (each features has its pubspec, lib/src, etc). Which might be a pro for some, or annoying for others. This is the reason I'm even googling another way.
 - It could be helpful to split arb files inside a package for cohesion.

What would be nice is simply using pattern matching to generate the classes. For example

```
/edit_profile
  /localization
    edit_profile.arb => generates the localization classes next to it.
/create_profile
  /localization
    create_profile.arb
```
